contrib/segmentio/kafka.go.v0/kafka_test.go

```diff
@@ -283,39 +447,42 @@ func TestNamingSchema(t *testing.T) {
 }
 
 func BenchmarkReaderStartSpan(b *testing.B) {
-	r := NewReader(kafka.ReaderConfig{
-		Brokers: []string{"localhost:9092", "localhost:9093", "localhost:9094"},
-		GroupID: testGroupID,
-		Topic:   testTopic,
-		MaxWait: testReaderMaxWait,
-	})
-
+	ctx := context.Background()
+	kafkaCfg := tracing.KafkaConfig{
+		BootstrapServers: "localhost:9092,localhost:9093,localhost:9094",
+		ConsumerGroupID:  testGroupID,
+	}
+	tr := tracing.NewTracer(kafkaCfg)
 	msg := kafka.Message{
 		Key:   []byte("key1"),
 		Value: []byte("value1"),
 	}
 
 	b.ResetTimer()
 	for n := 0; n < b.N; n++ {
-		r.startSpan(nil, &msg)
+		tr.StartConsumeSpan(ctx, wrapMessage(&msg))
```

Comment:
So this isn't your doing; but those functions return a span that is immediately dropped on the floor... and that makes this benchmark susceptible to have the compiler optimize away a bunch of side-effects that are rendered void by this.

Usually, one wants to assign the output to a package-scoped variable so the compiler doesn't do these optimizations. It would also be good form to clean up (read: finish) those spans.

Reply:
wow thanks, good to know this :)